### ISSUE TITLE
Lint sections with examples in all rules

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint.io#16]
Create markdown extraction plugin and lint all section with example in README.md in each rule. This will help reduce the number of errors in the documentation.

**Comment by jeddy3** [comment]
I'm not sure how worthwhile this is as the purpose of the examples in the documentation is to show infractions. So, we can lint them to a common standard as there isn't one.

**Comment by ntwb** [comment]
> So, we can lint them to a common standard as there isn't one.Thinking out loud....

I agree with this, initially I thought what I've added below would be how we could do it, but in the end these tests are really not testing anything, they are either a) testing the rule and that's what our tests already do or b) ignoring the rule and linting with an empty configuration.

____

If we were to test by linting each example, each example should only return 1 warning/error correct?

For example: https://github.com/stylelint/stylelint/tree/master/src/rules/no-extra-semicolons

Testing `no-extra-semicolons` with the option `true` then the following example would return 1 warning;

```css
@import "x.css";;
```

If we were to _extract_ that example for the purposes of linting these examples then utilising `stylelint-disable` with the same configuration would return no warnings or errors, e.g.

```css
/* stylelint-disable no-extra-semicolons */
@import "x.css";;
/* stylelint-enable */
```

Now with what I outlined above a _common standard_ wouldn't be required as each rule would have its own configuration, in the simplest form each rule would have only 1 rule in it's configuration. 

(For rules with multiple options or optional options then a new configuration for each section would contain the appropriate configuration for that section.)
